src/main/java/seedu/tp/group/FlashcardGroup.java

```diff
@@ -0,0 +1,90 @@
+package seedu.tp.group;
+
+import seedu.tp.exceptions.InvalidFlashcardIndexException;
+import seedu.tp.flashcard.Flashcard;
+import seedu.tp.flashcard.FlashcardList;
+import seedu.tp.ui.Ui;
+
+import static seedu.tp.utils.Constants.DESCRIPTION_FIELD;
+import static seedu.tp.utils.Constants.INDEXES_FIELD;
+import static seedu.tp.utils.Constants.NAME_FIELD;
+
+/**
+ *  A group of flashcards which have some of the same characteristics.
+ */
+public class FlashcardGroup {
+    private String name;
+    private String description;
+    private FlashcardList groupCards = new FlashcardList();
+
+    /**
+     * Constructs a <code>FlashcardGroup</code> using some existing cards from the users original list.
+     *
+     * @param name the name of the group.
+     * @param description brief description of the group.
+     * @param originalList the original list of flashcards.
+     * @param indexes the indexes of the flashcards which are going to be added to this group.
+     */
+    public FlashcardGroup(String name, String description,FlashcardList originalList,int[] indexes) {
+        this.name = name;
+        this.description = description;
+        for (int i : indexes) {
+            groupCards.addFlashcard(originalList.getFlashcardAtIdx(i));
+        }
+    }
+
+    /**
+     * Adds a flashcard to an existing group.
+     *
+     * @param flashcard the flashcard to be added.
+     * @return the flashcard be added.
+     */
+    public Flashcard addFlashcardToTheGroup(Flashcard flashcard) {
```

Comment:
Hmmmm why do we need to return the Flashcard added to the group since we do not make any changes to the flashcard in the method?